app/state_machines/flood_risk_engine/step_machine.rb

```diff
@@ -21,15 +21,17 @@ def current_step
       state.to_s
     end
 
+    # TODO: rename this method to something better. Style guide recommends
+    # not using set_ at start of method
+    # rubocop:disable Style/AccessorMethodName
     def set_step_as(step)
       restore!(step.to_sym)
     end
+    # rubocop:enable Style/AccessorMethodName
```

Comment:
The methods triggering things trigger events, not steps. An event causes the state to change (or at least be reassessed) which is then reflected in the step.

However, there is something nice and clean in `step=` when you are forcing the step to a certain state. So, I agree. That's the best solution so far.